app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_ingredients = @recipe.ingredients
+    @total_cost = @recipe.total_cost_ingredients
```

Comment:
These 2 instance variables are not really necessary; you can call `.ingredients` and `total_cost_ingredients` from the view itself, which would help keep this controller method "skinny".